src/DFA.cc

```diff
@@ -287,7 +285,7 @@ DFA_State* DFA_State_Cache::Lookup(const NFA_state_list& nfas, DigestStr* digest
     // HashKey because the data is copied into the key.
     hash128_t hash;
     KeyedHash::Hash128(id_tag, p - id_tag, &hash);
-    *digest = DigestStr(reinterpret_cast<const unsigned char*>(hash), 16);
+    *digest = DigestStr(reinterpret_cast<const char*>(hash), 16);
```

Comment:
Nothing to do - I guess it would be nice to be able to use `hash128_t` directly within a `std::map`, then we wouldn't even need the string. But that requires more surgery :-)

Reply:
Yeah I saw that too, there's a lot of cleanup for these files just scanning through so I only did the low hanging fruit in the second commit (which, admittedly, wasn't much. A lot of narrowing conversions are around here though, too)